src/JWT/Builder/JwtBuilder.cs

```diff
@@ -304,8 +332,31 @@ private bool CanDecode()
                 _dateTimeProvider != null &&
                 _urlEncoder != null)
             {
-                return !_verify || (_algorithm?.IsAsymmetric ?? true) || !String.IsNullOrEmpty(_secret);
+                return !_verify || (_algorithm?.IsAsymmetric ?? true) || HasSecrets();
             }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Checks if any secret was supplied to use in token decoding
+        /// </summary>
+        private bool HasSecrets()
+        {
+            if (_secrets != null && _secrets.Length > 0)
+                return true;
+
+            return false;
+        }
+
+        /// <summary>
+        /// Checks if there is only one secret was supplied for token encoding
+        /// </summary>
+        /// <returns></returns>
+        private bool HasOnlyOneSecret()
```

Comment:
Maybe HasSecret, HasSecrets? Or not raelly readable?

Reply:
I thought I would use some readable name, which will identify the purpose of the function

Reply:
Sure, sounds reasonable.